cmd/kubelet/app/server.go

```diff
@@ -1283,3 +1293,28 @@ func newTracerProvider(s *options.KubeletServer) (oteltrace.TracerProvider, erro
 	}
 	return tp, nil
 }
+
+func getCgroupDriverFromCRI(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Dependencies) error {
+	klog.V(4).InfoS("Getting CRI runtime configuration information")
+	runtimeConfig, err := kubeDeps.RemoteRuntimeService.RuntimeConfig(ctx)
+	if err != nil {
+		s, ok := status.FromError(err)
+		if !ok || s.Code() != codes.Unimplemented {
+			return err
+		}
+		// CRI implementation doesn't support RuntimeConfig, fallback
+		klog.InfoS("CRI implementation should be updated to support RuntimeConfig when KubeletCgroupDriverFromCRI feature gate has been enabled. Falling back to using cgroupDriver from kubelet config.")
```

Comment:
For beta, maybe we can even record an Event into the API?

Reply:
Might be good for the user. The proposal now states that for beta *Drop fallback to old behavior. CRI implementations expected to have support*. We can of course change that, or then add an event for alpha already 🧐